[bzip2] Add SHA512 hash check

StarGate-One opened the pull request:
**Describe the pull request**
1. Add SHA512 hash check to portfile.cmake
2. Delete old style CONTROL file
3. Add new style vcpkg.json file

- What does your PR fix? Fixes #13224 ([bzip2] Update to 1.0.8. #13178 - missing SHA512 hash check)
  
- Which triplets are supported/not supported? Changes made should not change triplets supported/not supported

- Have you updated the CI baseline? Changes made should not effect CI baseline

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes, it should

StarGate-One committed:
Add SHA512 Hash check to bzip2 portfile.cmake
Delete old style CONTROL file
Add new style vcpkg.json file

StarGate-One committed:
Update homepage for port bzip2 in vcpkg.json

c72578 commented on ports/bzip2/vcpkg.json:
```diff
@@ -0,0 +1,7 @@
+  "version-string": "1.0.8",
+  "port-version": 1,
+  "description": "High-quality data compressor.",
+  "homepage": "http://www.bzip.org/"
```
Please use this homepage, because http://www.bzip.org is not up-to-date:
```suggestion
  "homepage": "https://sourceware.org/bzip2/"
```

StarGate-One replied:
Because bzip.org is the homepage and spourceware is a download page... bzip.org gives the history and description and sourceware has the downloads.

c72578 replied:
@StarGate-One Please have a look at:
**bzip.org changes hands**
https://lwn.net/Articles/762264/

StarGate-One replied:
I updated, but sourceware.org needs to file for an updated SPDX license expression as I had to use "bzip2-1.0.6" for vcpkg to accept it.

StarGate-One committed:
Update vcpkg.json with new homepage, documentation and license links.

StarGate-One committed:
Change license to SPDX License Header

StarGate-One committed:
Change license to SPDX License Identifier

StarGate-One committed:
I hope this is the SPDX license expression

StarGate-One committed:
Remove license tag from vcpkg.json - it does not seem to work in all triplets

StarGate-One committed:
Upgrade geotrans to version 3.8
1. Remove old style Control file
2. Add new style vcpkg.json file
3. Update CI to reflect platforms not valid:
   geotrans:arm-uwp=fail
   geotrans:arm64-uwp=fail
   geotrans:arm64-windows=fail
   geotrans:x64-osx=fail
   geotrans:x64-uwp=fail
   geotrans:x64-windows-static=fail

StarGate-One committed:
I forgot port-version in vcpkg.json file, left it empty changed to 1

StarGate-One commented:
Somehow both my port changes got mixed together - closing starting over :(
